app/grandchallenge/algorithms/views.py

```diff
@@ -342,41 +393,130 @@ def get_success_url(self):
             kwargs={"slug": self.kwargs["slug"], "pk": self.object.pk},
         )
 
-    def get_remaining_jobs(self, *, credits_per_job: int,) -> Dict:
-        """
-        Determines the number of jobs left for the user and when the next job can be started
 
-        :return: A dictionary containing remaining_jobs (int) and
-        next_job_at (datetime)
-        """
-        now = timezone.now()
-        period = timedelta(days=30)
-        user_credit = Credit.objects.get(user=self.request.user)
+class AlgorithmExperimentCreate(
+    UserFormKwargsMixin,
+    LoginRequiredMixin,
+    ObjectPermissionRequiredMixin,
+    FormView,
+    RemainingJobsMixin,
+):
+    form_class = AlgorithmInputsForm
+    template_name = "algorithms/algorithm_inputs_form.html"
+    permission_required = (
+        f"{Algorithm._meta.app_label}.execute_{Algorithm._meta.model_name}"
+    )
+    job = None
```

Comment:
Is this needed?

Reply:
Yes, otherwise we get warnings "instance attribute job defined outside __init__". The view doesn't have a model, it's jsust a FormView